overlord/servicestate/service_control_test.go

```diff
@@ -108,6 +109,224 @@ func (s *serviceControlSuite) mockTestSnap(c *C) {
 	c.Assert(err, IsNil)
 	err = ioutil.WriteFile(filepath.Join(dirs.GlobalRootDir, "etc/systemd/system/snap.test-snap.foo.service"), nil, 0644)
 	c.Assert(err, IsNil)
+
+	return inf
+}
+
+func verifyControlTasks(c *C, tasks []*state.Task, expectedAction, expectedFlag string, expectedServices ...string) {
+	// sanity, ensures test checks below are hit
+	c.Assert(len(tasks) > 0, Equals, true)
+	var i int
+	for i < len(tasks) {
+		var argv []string
+		if tasks[i].Kind() == "exec-command" {
+			switch expectedAction {
+			case "start":
+				if expectedFlag != "" {
+					c.Assert(tasks[i].Get("argv", &argv), IsNil)
+					c.Check(argv, DeepEquals, append([]string{"systemctl", "enable"}, expectedServices...))
+					i++
+					wt := tasks[i].WaitTasks()
+					c.Assert(wt, HasLen, 1)
+					c.Assert(wt[0].ID(), Equals, tasks[i-1].ID())
+				}
+				c.Assert(tasks[i].Get("argv", &argv), IsNil)
+				c.Check(argv, DeepEquals, append([]string{"systemctl", "start"}, expectedServices...))
+			case "stop":
+				if expectedFlag != "" {
+					c.Assert(tasks[i].Get("argv", &argv), IsNil)
+					c.Check(argv, DeepEquals, append([]string{"systemctl", "disable"}, expectedServices...))
+					i++
+					wt := tasks[i].WaitTasks()
+					c.Assert(wt, HasLen, 1)
+					c.Assert(wt[0].ID(), Equals, tasks[i-1].ID())
+				}
+				c.Assert(tasks[i].Get("argv", &argv), IsNil)
+				c.Check(argv, DeepEquals, append([]string{"systemctl", "stop"}, expectedServices...))
+			case "restart":
+				if expectedFlag != "" {
+					c.Assert(tasks[i].Get("argv", &argv), IsNil)
+					c.Check(argv, DeepEquals, append([]string{"systemctl", "reload-or-restart"}, expectedServices...))
+				} else {
+					c.Assert(tasks[i].Get("argv", &argv), IsNil)
+					c.Check(argv, DeepEquals, append([]string{"systemctl", "restart"}, expectedServices...))
+				}
+			default:
+				c.Fatalf("unhandled action %s", expectedAction)
+			}
+		} else {
+			c.Fatalf("unexpected task: %s", tasks[i].Kind())
+		}
+		i++
+	}
+}
+
+func makeControlChange(c *C, st *state.State, inst *servicestate.Instruction, info *snap.Info) *state.Change {
+	apps := []*snap.AppInfo{}
+	for _, name := range inst.Names {
+		c.Assert(info.Apps[name], NotNil)
+		apps = append(apps, info.Apps[name])
+	}
+	st.Unlock()
+	tss, err := servicestate.Control(st, apps, inst, nil)
+	st.Lock()
+	c.Assert(err, IsNil)
+
+	chg := st.NewChange("service-control", "...")
+	for _, ts := range tss {
+		chg.AddAll(ts)
+	}
+	return chg
+}
+
+func (s *serviceControlSuite) TestControlConflict(c *C) {
+	st := s.state
+	st.Lock()
+
+	inf := s.mockTestSnap(c)
```

Comment:
We usually use the identifier `info` for this.

Reply:
Renamed.